core/packages/contracts/src/InboundQueue.sol

```diff
@@ -4,29 +4,27 @@ pragma solidity ^0.8.19;
 import {MerkleProof} from "openzeppelin/utils/cryptography/MerkleProof.sol";
 import {AccessControl} from "openzeppelin/access/AccessControl.sol";
 import {IParachainClient} from "./ParachainClient.sol";
+import {Registry} from "./Registry.sol";
+import {RegistryLookup} from "./RegistryLookup.sol";
+import {Auth} from "./Auth.sol";
+import {Vault} from "./Vault.sol";
+
 import {IRecipient} from "./IRecipient.sol";
-import {IVault} from "./IVault.sol";
 import {ParaID} from "./Types.sol";
 
-contract InboundQueue is AccessControl {
+contract InboundQueue is Auth, RegistryLookup {
```

Comment:
Should it be inherited from `Registry` or `Auth`?

Reply:
It needs to inherit from both (multiple inheritance). `Auth.sol` and `RegistryLookup` offer different capabilities to `InboundQueue`.